src/ziggurat/resource/dead_set.clj

```diff
@@ -30,21 +30,21 @@
   (some? channel))
 
 (defn retry-enabled? []
-  (get-in-config [:retry :enabled]))
+  (get-in (retry-config) [:enabled]))
 
-(defn channel-retry-enabled? [topic-entity stream-routes channel]
-  (get-in stream-routes [(keyword topic-entity) (keyword channel) :retry :enabled]))
+(defn channel-retry-enabled? [topic-entity channel]
+  (get-in (channel-retry-config (keyword topic-entity) (keyword channel)) [:enabled]))
 
-(defn retry-allowed? [topic-entity stream-routes channel]
+(defn retry-allowed? [topic-entity channel]
   (if (channel-request? channel)
-    (channel-retry-enabled? topic-entity stream-routes channel)
+    (channel-retry-enabled? topic-entity channel)
     (retry-enabled?)))
 
 (defn get-replay []
   (let [stream-routes (:stream-routes (mount/args))]
```

Comment:
we are getting the config from (mount/args) here. This breaks the api when people have `mount/start` in their `start-fn`. Since you have moved the `retry-allowed?` function away from this. I think it makes sense to move `validate-params` away as well.

Reply:
But it is different. `retry-allowed?` will check the configuration whether it is enabled or not. `validate-params` will check to parameters which are passed to ziggurat main function